### ISSUE TITLE
GL_INVALID_ENUM on Android

### ISSUE THREAD

**Issue by rubenwardy** [minetest/minetest#7525]
Reported by `NiklasGamerHD ✔` on [Discord](https://discord.gg/5XEgnjt)

##### Minetest version
```
0.4.17.2
```

##### OS / Hardware
Operating system: Android 7.0
Model: Huawei Honor 6A
GPU model: unknown
OpenGL version: unknown

![image](https://user-images.githubusercontent.com/2122943/42139258-3d78d8c8-7d82-11e8-9d42-3f959bb8e5f7.png)

![image](https://user-images.githubusercontent.com/2122943/42139250-2c89ba82-7d82-11e8-9ac5-38f1734f7f4c.png)

**Comment by rubenwardy** [comment]
Okay, so now they said this:

> I installed a font and then I tried to mince it but then overwrote something on a path

**Comment by stujones11** [comment]
I've not come across this in any of my tests, however, it is clear that there are some mods installed here. I can see some armor in there, though I'm pretty sure I fixed those iCCP warning some time ago.

**Comment by SmallJoker** [comment]
Same issue on Windows: https://forum.minetest.net/viewtopic.php?f=6&t=16243 (Outdated driver, wrong textures)
Android issue: https://forum.minetest.net/viewtopic.php?f=6&t=20091
Possible duplicate of #3080

**Comment by MoNTE48** [comment]
@rubenwardy this is CONFIRMED bug.
A simple solution is to **disable logs from Irrlicht** on Android.
None of Minetest developers will not correct this bug.
Right solution - hmm... The problem is at least 4 years. Hardly it will be decided soon. But let the Android players (especially on Android 8. +) will experience problems in the game. Nobody cares.

**Comment by notanewbie** [comment]
How do you disable the logs in Android?

**Comment by stujones11** [comment]
I came across this old topic but it is possibly still relevant. http://irrlicht.sourceforge.net/forum/viewtopic.php?t=43774

I agree with @MoNTE48 we should at least include an option to disable the Irrlicht logs. We can always patch Irrlicht if the problem lies there, however, from what I can gather it seems more likely a device/vendor specific issue.

**Comment by MoNTE48** [comment]
Simple solution:
https://github.com/MultiCraftProject/MultiCraft/commit/968954a3cc8dbdb818813ecd9c8476ff94447f7d#diff-4454bb181ab2b8fd8779df3a60282769R117

**Comment by codexp** [comment]
I confirm this bug happening on Samsung Galaxy S3 Tablet

**Comment by codexp** [comment]
The strange thing is that it has appeared recently and minetest has been working fine until it began throwing those errors. So something has changed? Is there any way to fix that without recompiling the app?

**Comment by MoNTE48** [comment]
@codexp 
just add this line on minetest.conf
debug_log_level = 
(withount any values)
Although, I'm not sure

**Comment by codexp** [comment]
Now I have this error on my PC. The only difference is that it will be logged only once on connect.
Whereas on Android it keeps flooding the console.

**Comment by MoNTE48** [comment]
@lhofhansl This is called **fork**, and not a _clone_, if you did not.
Disabling the Internet will save you from advertising

**Comment by jastevenson303** [comment]
Then forced to play single player?

**Comment by MoNTE48** [comment]
@jastevenson303, in next major release ads in singleplayer will be disabled

**Comment by codexp** [comment]
@MoNTE48 do you have paid version without ads?

**Comment by MoNTE48** [comment]
@codexp yes, $0.99
_but this is offtop in minetest repo!
Here I just offered my corrections in the minetest._

**Comment by aerozoic** [comment]
My phone does the same thing, constant flooding of this error.
Android 8.0
MT 0.4.17.2
![screenshot_20180721-192433](https://user-images.githubusercontent.com/15715378/43040849-f94b9b22-8d1c-11e8-9e4b-ba54c20d97cf.png)

**Comment by rubenwardy** [comment]
An anonymous source experienced with Graphics said this:

> If you want to fix GL_INVALID_ENUM errors, use KHR_debug extension with synchronous mode
> you will get callstacks
> if you put breakpoint inside your debugger
> in the callback

**Comment by paramat** [comment]
Does it happen on 5.0.0-dev also?

**Comment by stujones11** [comment]
If this is the same problem CuteAlien is referring to on the Irrlicht forums (link above) then our options could be quite limited. From what I can understand, some drivers may be falsely reporting the existence of certain extensions, hence why this only occurs on some devices.

Since this is now labelled as a blocker I can only suggest one of the following

1. Do not dump these messages to the game screen (android only)

- Pros: Fixes the user-facing issue fairly easily. (I think)

- Cons: Possible loss of useful debug information from users.

2. Suppress this particular error by directly patching Irrlicht during the build process. (already done for other issues)

- Pros: You still receive all other log messages.

- Cons: More Irrlicht patchwork and maintenance, also a dirty hack.

What seems clear to me is that it’s unlikely to be a problem that can be fixed within the engine itself, however, I hope I can be proved wrong about that :)

**Comment by codexp** [comment]
how about grouping same error messages as chrome console does?

**Comment by stujones11** [comment]
> Option 1 is also a dirty hack

I do not consider this to be a hack so long as it is kept optional and by this I do not mean to disable debug logging altogether, only the output to the game window. These and other non-fatal errors are not exclusive to android either, in many cases the best advice we can give it that the error/warning can be safely ignored, unfortunately that is not a lot of comfort to a mobile user.

**Comment by codexp** [comment]
I think that chat in multiplayer game plays a central role. If you are unable to communicate or it becomes very hard to do, a lot of usual actions in game become pointless. So this is not only a matter of comfort, but a serious issue.

**Comment by aerozoic** [comment]
It should be said, this issue makes multiplayer virtually useless to a huge amount of people. I would guess this has caused many people to switch to multicraft. I think this deserves high priority and should be fixed ASAP!
:large_orange_diamond:

**Comment by rubenwardy** [comment]
I think it would be better to fix this right than to fix it wrong. I suggest the following:

1. Hotfix and release 0.4.17.3 to hide the error message
2. Don't add the hotfix to 5.0, instead find the source of the issue and fix it properly. This may be causing performance issues

**Comment by stujones11** [comment]
For 1 I agree, that should be done asap. For 2 it will take someone who is able to reproduce this to properly trace it, however, I do not get the `GL_INVALID_ENUM` error on my (Samsung) devices. Unfortunately, there is nothing much we can do about the `iCCP` warnings other than tell server owners to fix their offending textures.

**Comment by rubenwardy** [comment]
I can't reproduce on my device either - I also have a Samsung.
I have a Sony device which I could potentially use if I could manage to actually turn it on

**Comment by MoNTE48** [comment]
This problem actually on Android 8.+, basically.

**Comment by rubenwardy** [comment]
I use Android 8.0

**Comment by paramat** [comment]
Please can we stop Android and desktop MT versions getting out of phase? It's getting confusing and messy.
This started when nerzhul named the current app store version 0.4.17.2, i warned against this at the time but it wasn't edited. Maybe it should have been 0.4.17.1.1 or something.
The next MT version is 0.4.17.2 not .3.

**Comment by paramat** [comment]
I agree with https://github.com/minetest/minetest/issues/7525#issuecomment-410245217 the app should be fixed ASAP even if slightly hacky.

**Comment by serofir** [comment]
This problem actually on Android 6,7,8.

**Comment by alnixon** [comment]
I was experiencing this issue with my Nexus 6p (running omnirom7.1.2) minetest version 0.4.16.17 from f-droid.
I tried wiping the data folder and installing 0.4.17.1-2 from github releases but it crashed immediately.
I the wiped the data folder again and tried 0.4.16 from github releases and everything works fine.

**Comment by paramat** [comment]
Bump, needs attention as is a blocker for 5.0.0.

**Comment by MoNTE48** [comment]
@stujones11 Do not make fun of me. Disabling lines with logs on Android is not code :)

**Comment by stujones11** [comment]
@MoNTE48 I was not making fun of you, just respecting your license but I guess this really does not count here. So long as you don't have a problem with us using it, that great :)

**Comment by paramat** [comment]
stujones11 i agree, the simple fix is good enough for the short term, doing the proper fix should not delay 5.0, and it can always be done later if someone finds time.
Please can anyone code the simple fix?

**Comment by stujones11** [comment]
@paramat it would simply be that section of code enclosed in a preprocessor directive.

https://github.com/minetest/minetest/blob/0e306c084284aeafb3d5cde0cfec11a85a11cb9c/src/client/inputhandler.cpp#L173

```c++
#if !defined(__ANDROID__) || !defined(NDEBUG)
		g_logger.log(irr_loglev_conv[event.LogEvent.Level],
				std::string("Irrlicht: ") + event.LogEvent.Text);
#endif
```

**Comment by paramat** [comment]
Ok.

**Comment by paramat** [comment]
stujones11 are you able to verify that yourself or do you need help with that?

**Comment by paramat** [comment]
See http://irc.minetest.net/minetest-dev/2018-11-03#i_5433285

**Comment by stujones11** [comment]
> are you able to verify that yourself or do you need help with that?

I think I should be able to do that, I just can't actually sign the release.  I will try some of those ideas.

**Comment by paramat** [comment]
Thanks, do you want to make the PR or shall i?
Rubenwardy prefers a proper fix be done for 5.0, which is understandable, but at the least any fix can be backported to 0.4.x.
Also, if merged into 5.0-dev this will temporarily help those testing Android, so i think it's worth merging even if temporary.

**Comment by stujones11** [comment]
Rubenwardy is correct that this could also be impacting performance for those affected and I will continue to look into this, however, I am only really guessing until I can reproduce this myself.

>We would keep this issue open to remind us to do the proper fix.

:+1:

**Comment by paramat** [comment]
Ok.

**Comment by paramat** [comment]
#7837

**Comment by stujones11** [comment]
As of the recent Irrlicht 1.9 updates, I am now able to get MT to build with the latest Irrlicht-ES sources. It may be a bit of a long-shot but there may be some chance that this issue no longer presents itself.

What is interesting is that two of our patch files no longer apply. One, `irrlicht-native_activity.patch`  is clearly now already applied upstream, the other `irrlicht-texturehack.patch` is for a file that no longer exists and I cannot even seem to find where the code has moved to.

It may well be that this patch is no longer required and there is some evidence to suggest that it could be linked to the `GL_INVALID_ENUM` error. Specifically this message seems to always accompany it.
https://github.com/minetest/minetest/blob/b78698240c4b0e4ac410e2aafe6d88b34b1618e2/build/android/patches/irrlicht-texturehack.patch#L50

A debug apk is available here for testing: https://www.dropbox.com/s/xfvhygfnsowkntn/Minetest-debug.apk?dl=0

I am particularly keen to hear from anyone that is able to reproduce this issue.

Feature branch with necessary `Makefile` changes here: https://github.com/stujones11/minetest/tree/android-irrlicht-1.9

**Comment by MoNTE48** [comment]
@stujones11 Why you download irrlicht from https://github.com/zaki/irrlicht/tree/ogl-es ? In this repo 1 year old code. You can pull code from official svn

**Comment by stujones11** [comment]
@MoNTE48 svn is very slow, at least for me this is much a quicker method of downloading the sources,

**Comment by MoNTE48** [comment]
@stujones11 special for you :D https://github.com/MoNTE48/irrlicht/tree/ogl-es
But now i don't know how update this repo :) I just import svn repo.

**Comment by stujones11** [comment]
@MoNTE48 Thanks, so much for that being an `automatically` updating mirror :(

I will make another build with the very latest sources as soon as I get the chance, although I doubt much will have changed in a year regarding the ogl-es branch but you never know.

**Comment by MoNTE48** [comment]
Stu, there's nothing wrong with open PR. Sooner or later they will pay attention. Just need to be checked by several people. I really appreciate your work, in reality the android port looks abandoned. Since I am no longer interested in developing MC, I plan to suggest some interesting changes.
Although it is very likely that after testing they will also be waiting for a long time :)

**Comment by stujones11** [comment]
> Stu, there's nothing wrong with open PR.

Sure, but I am not a bit a huge fan of rebasing and merge conflict resolving :D

> I plan to suggest some interesting changes.

I look forward to seeing these. It seems a shame you are abandoning MultiCraft, it was probably the most legitimate fork we had. I have no idea what the problem was but also off-topic here :)

**Comment by MoNTE48** [comment]
@stujones11 your apk from this msg https://github.com/minetest/minetest/issues/7525#issuecomment-437703583 NOT fixed problem, i'm sorry. Maybe, 
> a year regarding the ogl-es branch
can help we :)

>  I have no idea what the problem was but also off-topic here :)

Google blocked my developer, wallet and admob accounts. I can't restore my 50+ millions downloads app or re-public new app. Also, for more than two months, they did not answer me about the reason for blocking. This shows that Google just doesn't care about you. Even if you are a "big" developer.

**Comment by stujones11** [comment]
> a year regarding the ogl-es branch can help

I will try asap but I am done for today :) you can always substitute the git url and commit hash in the `Makefile` and try it. It seems doubtful it will help though :(

> for more than two months, they did not answer me
> Google just doesn't care about you.

That sucks dude, sorry to hear that!

**Comment by paramat** [comment]
This is shocking news you are no longer developing MC.
Looks like we really will need to sort out the official app for 5.0.0 since there will be a lot of players moving to another app.
Did you eventually get informed why you were blocked? If so why was that, if you don't mind sharing that?

**Comment by notanewbie** [comment]
I get a "Can't Open File" error when I try to open the debug APK.

**Comment by stujones11** [comment]
> I get a "Can't Open File" error when I try to open the debug APK.

You may need to enable 'unknown sources', for Android 8 it's slightly different https://stackoverflow.com/questions/46005766/installing-apk-cannot-open-file-on-android-8-0

@MoNTE48 FWIW, I have made a build using the latest sources from your svn import, it certainly seems to build a run ok for me. I have updated the link here https://github.com/minetest/minetest/issues/7525#issuecomment-437703583 if you care to try it.

However, I am not very optimistic it well help in this situation. To me, this looks increasingly likely to be a vendor driver bug. Essentially, the driver falsely reports the existence of a particular extension which Irrlicht then attempts to use resulting in the invalid enumerator error.

**Comment by MoNTE48** [comment]
@stujones11 Yes you are right... Correcting this error will be more difficult than using the latest Irrlicht... Perhaps a logical solution would be to make a patch that turns off this particular error in the Irrlicht itself, and does not disable the display of all errors on Android?

**Comment by stujones11** [comment]
@MoNTE48 I agree, this would better than disabling all Irrlicht messages. I might see if I can up come with a patch that simply demotes that error to a warning, that way it will still be logged but not spam the game screen (I think anyway)

**Comment by MoNTE48** [comment]
The most important thing is not create biggest debug.txt...

**Comment by stujones11** [comment]
> The most important thing is not create biggest debug.txt...

I agree to some extent, however, I think this is a slightly different issue. MT debug.txt can get very large regardless of this particular error. There probably should be some type of 'auto-cleaning' mechanism of some kind. Perhaps overwrite just the old one on each new session.

**Comment by MoNTE48** [comment]
Yes, because it can become quite large through a large number of games. I'm not sure, but adding a line in txt over 5 MB in size is a resource-intensive task for slow emmc and weak cpu?

**Comment by stujones11** [comment]
> 5 MB in size is a resource-intensive task for slow emmc and weak cpu?

Time stamping them on each new session would help in this regard but would still fill up the internal storage over time. But that is another possible solution.

**Comment by paramat** [comment]
Looks like several messages per second, i'd rather these were not logged anywhere.
>  I might see if I can come up with a patch that simply demotes that error to a warning

Do you mean there may be a practical way to detect that specific error? If so, just don't log it at all, to avoid filling up debug.txt with a useless message.
How about something like:
if 'error' then
if stringmatch 'invalid enum' then
don't log
?

**Comment by stujones11** [comment]
> Do you mean there may be a practical way to detect that specific error?

Certainly it you patch Irrlicht directly during the build process, that is what I thinking of doing. However, your idea should work equally well and would allow you to perhaps log the error only once? Patching would be slightly more efficient and it saves us from adding yet more android hacks to the MT codebase but I am neutral on the implementation.

**Comment by paramat** [comment]
I might update my PR in case patching Irrlicht does not happen for some reason. At the least it will be good enough for a temporary fix for 0.4.17.

**Comment by stujones11** [comment]
@paramat Patching irrlicht is nowhere near as difficult as it may sound. However, I just made a very interesting discovery while looking to do this. The only place where this error is shown is guarded by `#ifdef _DEBUG` directive. Unless I am mistakes, it looks like we might be building irrlicht in debug mode? We must have been doing so for quite some time if that is the case. I need to investigate this further

From the latest sources, here: https://github.com/MoNTE48/irrlicht/blob/5b8b895cca0cf59bffea6564dd5d6c3e80947d8d/source/Irrlicht/COGLESDriver.cpp#L1442

I cannot show you the exact lines of the 2015 version we are currently using but little has changed.

**Comment by paramat** [comment]
Weird. @nerzhul is it possible that the Irrlicht used for the playstore app is in debug mode?

**Comment by paramat** [comment]
Well, MoNTE48 says this is a years old bug so i doubt it and hope not.
Patching Irrlicht seems the ideal solution for 5.0, whoever creates the app store release would build using the patched Irrlicht. However that means building from the MT repo would be more difficult. Maybe the MT repo could contain a patched fork of Irrlicht with just that one change?

**Comment by rubenwardy** [comment]
> Patching Irrlicht seems the ideal solution for 5.0, whoever creates the app store release would build using the patched Irrlicht. However that means building from the MT repo would be more difficult. Maybe the MT repo could contain a patched fork of Irrlicht with just that one change?

This is already possible, and is done 4 times already.

https://github.com/minetest/minetest/tree/master/build/android/patches

**Comment by stujones11** [comment]
@MoNTE48 Could you please test this unsigned release build if you are able.

https://www.dropbox.com/s/nxcacu7rs520vkv/Minetest-release-unsigned.apk?dl=0

**Comment by stujones11** [comment]
@paramat What we need to figure out is why `_DEBUG` is defined, that code should never be executed in the first place. It is obvious why my debug builds would show this but not the play-store release!

**Comment by paramat** [comment]
I agree, needs checking.
Good to see patching Irrlicht is already supported in MT.

**Comment by MoNTE48** [comment]
@stujones11 thanks, now it working for me.
I don't look any irrlicht errors in chat!

**Comment by stujones11** [comment]
Clearly then @nerzhul is incompetent :P

**Comment by MoNTE48** [comment]
@stujones11 yes)))
Do you plan to create PR?

**Comment by stujones11** [comment]
@MoNTE48 There is no PR, just `make release -j$(nproc)` it could not get easier.

I have absolutely no idea how he managed to do this, unless the whole apk is in debug mode also :D

**Comment by MoNTE48** [comment]
@stujones11 
I think that nrz build so:
```
git clone ...
make debug
```
and after some time
`make release`
Because of this, libraries (for example, irrlicht) were not re-make and remained compiled with debugging enabled.

**Comment by stujones11** [comment]
Or maybe he did not back-port my build fixes properly, I didn't really check.

**Comment by paramat** [comment]
So to clarify, the current playstore app was built with Irrlicht in debug mode?

Good news, @nerzhul please could you do a point release in the play store to fix this?

**Comment by nerzhul** [comment]
@stujones11 ok i'm incompetent, nice

**Comment by nerzhul** [comment]
then what is to do, i don't see any change to apply there

**Comment by stujones11** [comment]
@nerzhul Just build it properly, in a clean environment. See: https://github.com/minetest/minetest/issues/7525#issuecomment-439225467

I will check the 0.4.17.x version when I get time but it is more likely what MoNTE48 wrote.

**Comment by rubenwardy** [comment]
I think it's important to make sure this doesn't happen again. I suggest making the builds using CI, as an unsigned release build. To make a play store release, this build can be downloaded from the CI and the signed locally using the key. This signed copy is then uploaded

Having CI Android builds would also be very useful

**Comment by paramat** [comment]
@nerzhul to be clear, the current playstore app 0.4.17.2 has this build mistake, so it needs to be fixed immediately as a point release.
There's nothing to do for 5.0 of course, except avoid the same thing happening.

**Comment by nerzhul** [comment]
@rubenwardy CI is possible but it take 1 hour to build on my machine, don't expect to be faster on travis :p

**Comment by MoNTE48** [comment]
Less than 5 minutes on Core i7-8700k :D

**Comment by stujones11** [comment]
@nerzhul the whole build for me takes about 10-15 minutes, including the download of the dep sources. Mostly inhibited by the Irrlicht svn download.

My PC is nothing special, well maybe 5 years ago decent spec

**Comment by nerzhul** [comment]
i build MT in 15 sec , but all deps are long because i'm not building them with +8 to prevent some compilation issues due to cross dep build.
Irrlicht SVN download is slower in france than you it seems, for me it's 10-15 min to download that subversion shit from sourceforgeshit.

**Comment by stujones11** [comment]
>  t's 10-15 min to download that subversion shit from sourceforgeshit.

The simple answer would be to create our own mirror of the Irrlicht-ES sources. Or even Irrlicht istelf, for that matter.

**Comment by paramat** [comment]
Note, due to #7813 we may need to merge #7820 before fixing the app, but not sure.

**Comment by nerzhul** [comment]
Please test the build provided in the release section of GH https://github.com/minetest/minetest/releases/download/0.4.17.1/Minetest-0.4.17.1-3.apk

**Comment by rubenwardy** [comment]
A user reports they're still getting the same error with that

**Comment by paramat** [comment]
#7820 merged, so SDK 26 is now targeted, which may be necessary for updating the app (?).

**Comment by stujones11** [comment]
Targeting SDK 26 is now mandatory for any new play-store release, so yes, it is necessary but there is no real need to backport it so long as the playstore release is compiled with these changes.

@nerzhul I might also suggest that you upload your release candidate to github releases first so I/we can verify it before you push to the play-store.

Also, given the circumstances, I think this you should be a higher priority than code-refactors for you right now. Our current app is unplayable for many with reduced performance for everyone else.

**Comment by nerzhul** [comment]
@stujones11 then we need to split the @MoNTE48 PR in multiple parts. First compat for play store with point release in stable-0.4 branch

**Comment by stujones11** [comment]
@nerzhul AFAIK those java changes and bug-fixes should work equally well with version 0.4.x, though I did not try this yet. I don't see any point in splitting them unless there is a need to.

**Comment by MoNTE48** [comment]
@rubenwardy why minetest official app public in play store on nrz dev account?
Minetest community can't one time pay $25 for official dev account?
On dev account we can download logcats, crash analytics and lots of useful statistics that nrz does not publish.

**Comment by rubenwardy** [comment]
> why minetest official app public in play store on nrz dev account?
> Minetest community can't one time pay $25 for official dev account?
 
+10000

I'm sure many people would be happy to pay for it, but you need co-operation from nerzhul to actually transfer the app

**Comment by MoNTE48** [comment]
> but you need co-operation from nerzhul

He-he, not me, **YOU**! :)
I don't think this is a problem. Unless _he_ is afraid of losing control. I think that **you** are much more experienced in Java than it is.

**Comment by paramat** [comment]
Nerzhul has shared the playstore access with 1 or 2 other core devs as far as i remember. However i agree it's best to have group access to it.

**Comment by nerzhul** [comment]
@MoNTE48 in fact i have a real android full app on the store and at work i'm hosting java applications & microservices produces by our agile development team and i help them to increase their code quality and architecture, trolling is easy, but i have more experience than you think :p

for the rights on the app on my account, @celeron55 , @sfan5 and @rubenwardy share the app ownership with me, i'm just the account owner. It was done on my account because i got it when it was published. I'm not sure if we can move it to another account, there is already a dead MT app in my account (wrong artifact name) and google refused to remove it...

**Comment by MoNTE48** [comment]
>  I'm not sure if we can move it to another account
**We can**

> java.lang.UnsatisfiedLinkError: dlopen failed: cannot locate symbol "stderr" referenced by "libgmp.so"...
We can disable libgmp for Android, in MC i remove this lib and had no difficulties, no usage scenarios, when it is really needed.

**Comment by ClobberXD** [comment]
> I can confirm that this issue still exists in nerzhul's `0.4.17.1-3.apk`.

I tried out stujones' unofficial APK, and can confirm that this issue is fixed in 5.0.0.

**Comment by rubenwardy** [comment]
> I tried out stujones' unofficial APK, and can confirm that this issue is fixed in 5.0.0-dev. I tried both singleplayer and multiplayer just to be sure

This will be because stujones1 is building it properly, the problem is from how it is built not the code

**Comment by nerzhul** [comment]
I build it properly too... i will provide a 5.0.0-dev APK asap and you will see if that works properly :)

**Comment by pgimeno** [comment]
This APK: https://github.com/minetest/minetest/releases/download/0.4.17.1/Minetest-0.4.17.3.apk fixed this problem for me (I was hit by this issue before).

While I'm here: I've set up another automatically updated mirror of the Irrlicht SVN repo (updated daily) here: https://gitlab.com/pgimeno/irrlicht-mirror

**Comment by ClobberXD** [comment]
I can confirm. The latest APK (0.4.17.1-3 I assume - the versioning is getting really confusing :P) fixes the GL_INVALD_ENUM error for me as well. Thanks a ton. :)
